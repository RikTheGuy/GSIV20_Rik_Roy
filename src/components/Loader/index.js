import React from 'react'
import './index.scss'

const index = () => {
    return (
        <div data-testid='loader' className='loader'>
            <div data-testid='spinner' className='spinner'></div >
            <div data-testid='text'>Loading</div>
        </div>
    )
}

export default index
