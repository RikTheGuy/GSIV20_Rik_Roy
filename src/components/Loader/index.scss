@import "../../styles/colors.scss";

.loader {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  text-align: center;

  .spinner {
    display: block;
    width: 5rem;
    height: 5rem;
    background-color: transparent;
    border: 1px solid $accent;
    box-shadow: 0 0 1rem $accent;
    border-bottom: 0;
    border-radius: 50%;
    animation: spinner-load 0.5s linear infinite;
  }

  > div:last-child {
    margin-top: 1rem;
  }

  @keyframes spinner-load {
    0% {
      transform: rotate(0deg);
    }
    50% {
      transform: rotate(180deg);
    }

    100% {
      transform: rotate(360deg);
    }
  }
}
