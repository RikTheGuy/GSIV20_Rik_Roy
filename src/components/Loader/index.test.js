import Loader from './index.js';
import { render, cleanup } from '@testing-library/react';

import renderer from 'react-test-renderer';

describe('Loader', () => {
    afterEach(cleanup)

    test('Renders Successfully', () => {
        const { getByTestId } = render(<Loader />)
        expect(getByTestId('loader')).toContainElement(getByTestId('spinner'))
        expect(getByTestId('loader')).toContainElement(getByTestId('text'))
        expect(getByTestId('text')).toHaveTextContent('Loading')
    })

    test('Snapshot Testing', () => {
        const tree = renderer.create(<Loader />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})