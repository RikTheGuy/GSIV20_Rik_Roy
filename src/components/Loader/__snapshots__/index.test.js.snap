// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Snapshot Testing 1`] = `
<div
  className="loader"
  data-testid="loader"
>
  <div
    className="spinner"
    data-testid="spinner"
  />
  <div
    data-testid="text"
  >
    Loading
  </div>
</div>
`;
